packages/@sanity/desk-tool/src/pane/EditorPane.js

```diff
@@ -368,6 +362,40 @@ export default withDocumentType(
         })
     }
 
+    handleRestoreRevision = restoredDocument => {
+      const documentId = this.props.options.id
+      this.setState({isRestoring: true})
+      const tx = client.observable.transaction()
+      tx.createOrReplace({
+        ...omit(restoredDocument, '_updatedAt'),
+        _id: getDraftId(documentId)
+      })
+      tx.commit()
+        .pipe(
+          map(result => ({
+            type: 'success',
+            result: result
+          })),
+          catchError(error =>
+            observableOf({
+              type: 'error',
+              message: 'An error occurred while attempting to restore the document',
+              error
+            })
+          )
+        )
+        .subscribe({
+          next: result => {
+            this.setState({
+              transactionResult: result
```

Comment:
This should either use `setStateIfMounted` or (perhaps preferably) set the observable to the class instance and unsubscribe it in `willUnmount`